clients/ts/signalr/src/NodeHttpClient.ts

```diff
@@ -33,7 +34,15 @@ export class NodeHttpClient extends HttpClient {
                 port: url.port,
             };
 
-            const req = http.request(options, (res: http.IncomingMessage) => {
+            let httpOrHttps = http;
+            if (request.url) {
+                // poor man's "startsWith()"
+                if (request.url.lastIndexOf("https", 0) === 0) {
```

Comment:
Check `url.protocol` instead. We already load the URL up into a `URL` so use it :).

```javascript
const httpOrHttps = url.protocol === "https:" ? https : http;
```